### PR TITLE
Passing cols or rows into ctor as undefined will break terminal

### LINKED ISSUE #3329: Passing cols or rows into ctor as undefined will break terminal
This was causing https://github.com/microsoft/vscode/pull/123502 to fail CI and the reason ended up being that cols/rows set to undefined breaks. `null` works fine, as does not setting any cols/rows property.

### PR DESCRIPTION
**Pull request by silamon**
https://github.com/xtermjs/xterm.js/blob/554f7dab724cf63ad7b0197304eabf37992b9a68/src/common/services/BufferService.ts#L39-L40
Passing `undefined` to cols or rows would result in NaN. Passing `null` works as expected. 

Fixes #3329

### CODE REVIEW

File: src/common/services/OptionsService.test.ts [hunk 1]
```diff
@@ -10,13 +10,23 @@ describe('OptionsService', () => {
   describe('constructor', () => {
     const originalError = console.error;
     beforeEach(() => {
-      console.error = () => {};
+      console.error = () => { };
```
Comment (meganrogge) [inline src/common/services/OptionsService.test.ts:13]: should undo this formatting change

File: src/common/services/OptionsService.test.ts [hunk 2]
```diff
@@ -10,13 +10,23 @@ describe('OptionsService', () => {
...
+      assert.equal(optionsService.getOption('cols'), DEFAULT_OPTIONS.cols);
+    });
+    it('uses values from constructor option values if correctly passed', () => {
+      const optionsService = new OptionsService({ cols: 80, rows: 25 });
+      assert.equal(optionsService.getOption('rows'), 25);
+      assert.equal(optionsService.getOption('cols'), 80);
+    });
     it('uses default value if invalid constructor option value passed', () => {
-      assert.equal(new OptionsService({tabStopWidth: 0}).getOption('tabStopWidth'), DEFAULT_OPTIONS.tabStopWidth);
+      assert.equal(new OptionsService({ tabStopWidth: 0 }).getOption('tabStopWidth'), DEFAULT_OPTIONS.tabStopWidth);
```
Comment (meganrogge) [inline src/common/services/OptionsService.test.ts:29]:
```suggestion
      assert.equal(new OptionsService({tabStopWidth: 0}).getOption('tabStopWidth'), DEFAULT_OPTIONS.tabStopWidth);
```

File: src/common/services/OptionsService.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export const DEFAULT_OPTIONS: ITerminalOptions = Object.freeze({
   cursorStyle: 'block',
   cursorWidth: 1,
   customGlyphs: true,
-  bellSound:  DEFAULT_BELL_SOUND,
```
Comment (meganrogge) [inline src/common/services/OptionsService.ts:25]: undo

File: src/common/services/OptionsService.ts [hunk 4]
```diff
@@ -128,7 +128,7 @@ export class OptionsService implements IOptionsService {
         break;
       case 'cursorWidth':
         value = Math.floor(value);
-        // Fall through for bounds check
```
Comment (meganrogge) [inline src/common/services/OptionsService.ts:131]: undo

**Review by meganrogge — CHANGES_REQUESTED** [review]
looks good, thanks - a few formatting changes should be reverted

**Review by Tyriar — APPROVED** [review]
Thanks!

**Pull request merged**